symphony/bdk/core/service/message/message_service.py

```diff
@@ -464,6 +464,36 @@ async def search_messages_one_page(skip, limit):
 
         return offset_based_pagination(search_messages_one_page, chunk_size, max_number)
 
+    @retry
+    async def update_message(self, stream_id: str, message_id: str, message: Union[str, Message], data=None,
+                             version: str = "") -> V4Message:
+        """Update an existing message. The existing message must be a valid social message, that has not been deleted.
+        See: `Update Message <https://developers.symphony.com/restapi/reference#update-message-v4>`_
+
+        :param stream_id: The ID of the stream where the message is being updated.
+        :param message_id: The ID of the message that is being updated.
+        :param message: a :py:class:`Message` instance or a string containing the MessageML content to be sent.
+          If it is a :py:class:`Message` instance, other parameters will be ignored.
+          If it is a string, ``<messageML>`` tags can be omitted.
+        :param data: an object (e.g. dict) that will be serialized into JSON using ``json.dumps``.
+        :param version: Optional message version in the format "major.minor".
+          If empty, defaults to the latest supported version.
+
+        :return: a V4Message object containing the details of the updated message.
+        """
+        message_object = message if isinstance(message, Message) else Message(content=message, data=data, version=version)
+
+        params = {
+            "sid": stream_id,
+            "mid": message_id,
+            "session_token": await self._auth_session.session_token,
+            "key_manager_token": await self._auth_session.key_manager_token,
+            "message": message_object.content,
+            "data": message_object.data,
+            "version": message_object.version
```

Comment:
uh, I just discovered this parameter... I think I've never seen it used 🙄

Reply:
Same here, I saw it was specified in the docs and sendMessage method so I used it.